workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/FullScreenModalFactory.kt

```diff
@@ -0,0 +1,25 @@
+package com.squareup.workflow1.ui.container
+
+import android.content.Context
+import androidx.activity.ComponentDialog
+import com.squareup.workflow1.ui.Screen
+import com.squareup.workflow1.ui.ViewEnvironment
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+
+/**
+ * Default [OverlayDialogFactory] for the standard [FullScreenModal] rendering class.
+ * Nothing more than a direct call to [ComponentDialog.setContent].
+ *
+ * To provide a custom binding for [FullScreenModal], see [OverlayDialogFactoryFinder].
+ */
+@WorkflowUiExperimentalApi
+internal class FullScreenModalFactory<C : Screen>() : OverlayDialogFactory<FullScreenModal<C>> {
+  override val type = FullScreenModal::class
+
+  override fun buildDialog(
+    initialRendering: FullScreenModal<C>,
+    initialEnvironment: ViewEnvironment,
+    context: Context
+  ): OverlayDialogHolder<FullScreenModal<C>> =
+    ComponentDialog(context).setContent(initialRendering, initialEnvironment)
```

Comment:
Oh cheeky (return value is the holder)

Reply:
What I do like from this approach is that other factories don't need to know about setting up the update lambda. Looks cleaner. Probably not an issue already. But nice.

Reply:
Maybe that name is not clear enough, though. Is `withScreenContent` better?

Reply:
@steve-the-edwards be interested in your take on the naming here.

Reply:
Maybe `asDialogHolderWithScreens`,

Reply:
Yup, that's a better name, but will make that change in a downstream PR.

Reply:
Agreed I like `asDialogHolderWithContent`!